docs/quickstarts/express.mdx

```diff
@@ -110,30 +112,44 @@ Learn how to integrate Clerk into your Express backend for secure user authentic
 
   To protect your routes, use the [`requireAuth()`](/docs/references/express/overview#require-auth) middleware. This middleware functions similarly to `clerkMiddleware()`, but also protects your routes by redirecting unauthenticated users to the sign-in page.
 
-  In the following example, `requireAuth()` is used to protect the `/protected` route. If the user is not authenticated, they are redirected to the '/sign-in' route. If the user is authenticated, the `req.auth` object is used to get the `userId`, which is passed to [`clerkClient.users.getUser()`](/docs/references/backend/user/get-user){{ target: '_blank' }} to fetch the current user's `User` object.
+  In the following example, `requireAuth()` is used to protect the `/protected` route. If the user is not authenticated, they are redirected to the homepage. If the user is authenticated, the [`getAuth()`](/docs/references/express/overview#get-auth) function is used to get the `userId`, which is passed to [`clerkClient.users.getUser()`](/docs/references/backend/user/get-user){{ target: '_blank' }} to fetch the current user's `User` object.
 
   ```ts {{ filename: 'index.ts', mark: [3, [7, 11], [13, 16]] }}
   import 'dotenv/config'
   import express from 'express'
   import { clerkClient, requireAuth } from '@clerk/express'
 
   const app = express()
+  const PORT = 3000
 
-  app.get('/protected', requireAuth({ signInUrl: '/sign-in' }), async (req, res) => {
-    const { userId } = req.auth
+  // Use requireAuth() to protect this route
+  // If user is not authenticated, requireAuth() will redirect back to the homepage
+  app.get('/protected', requireAuth(), async (req, res) => {
+    // Use `getAuth()` to get the user's `userId`
+    const { userId } = getAuth(req)
+
+    // Use Clerk's JavaScript Backend SDK to get the user's User object
     const user = await clerkClient.users.getUser(userId)
-    return res.json({ user })
-  })
 
-  app.get('/sign-in', (req, res) => {
-    // Assuming you have a template engine installed and are using a Clerk JavaScript SDK on this page
-    res.render('sign-in')
+    return res.json({ user })
   })
 
-  app.listen(3000, () => {
+  // Start the server and listen on the specified port
+  app.listen(PORT, () => {
     console.log(`Example app listening at http://localhost:${PORT}`)
   })
   ```
+
+  ## Add global TypeScript type (optional)
+
+  If you're using TypeScript, you can add a global type to your project to provide auto-completion and prevent TypeScript errors.
```

Comment:
What if we made it more specific, something like

"If you're using TypeScript, you can add a global type reference to your project to enable auto-completion and type checking when accessing the `auth` object in Express request handlers."

Reply:
https://github.com/clerk/clerk-docs/pull/1899/commits/c23ff3031ca3e18b65b1da1674eab32e74af4fea